File: extensions/real-estate/front-end/Web/packages/platform/src/views/Command/Occupancy/Occupancy.js
import { Fetch, Flex } from '@willow/ui'
import Floors from './Floors/Floors'
import Report from './Report/Report'

export default function Occupancy() {
  return (
    <Fetch
      url="/api/pilot/occupancy"
      params={{
        hasBaseModule: true,
      }}
    >
      {(floors) => (
        <Flex
          horizontal
          fill="equal"
          height="100%"
          size="small"
          padding="small"
        >
          <Floors
            floors={floors.map((floor) => ({
              ...floor,
              id: floor.floorId,
              name: floor.floorName,
              code: floor.floorCode,
              people: Math.max(floor.runningTotal ?? 0, 0),
              peopleLimit: floor.floorLimit,
            }))}
          />
          <Report />
        </Flex>
      )}
    </Fetch>
  )
}

File: extensions/real-estate/back-end/PlatformPortalXL/src/PlatformPortalXL/Models/Occupancy.cs
using System;

namespace PlatformPortalXL.Models
{
    public class Occupancy
    {
        public Guid FloorId { get; set; }
        public string FloorName { get; set; }
        public string FloorCode { get; set; }
        public int? RunningTotal { get; set; }
        public int? FloorLimit { get; set; }
        public Guid ReportId { get; set; }
    }
}
